addon/utils/dom-helpers.ts

```diff
@@ -442,6 +443,119 @@ function getWindowSelection(): Selection {
     );
   return selection;
 }
+/**
+ * Gets a CaretPosition from a set of view coordinates.
+ * It will find the deepest node at that point, and give you
+ * the node and the offset which you need to put the caret as close to that point as possible,
+ * while still maintaining a valid caret position. Chrome and Firefox disagree on its
+ * interface, but the polyfill is trivial.
+ * */
+function getCaretPositionFromPoint(x: number, y: number): CaretPosition | null {
+  if (!document.caretPositionFromPoint) {
+    var range = document.caretRangeFromPoint(x, y);
+    if (range === null) return null;
+    return {
+      offset: range.startOffset,
+      offsetNode: range.startContainer,
+      getClientRect: range.getBoundingClientRect,
+    };
+  }
+  return document.caretPositionFromPoint(x, y);
+}
+/**
+ * Place the caret at point x, y. Caret will be positioned inside the deepest node at that point
+ * at an offset which puts it closest to the x,y point
+ * */
+function setCaretOnPoint(x: number, y: number) {
+  const caretPos = getCaretPositionFromPoint(x, y);
+  if (!caretPos) {
+    console.warn(`Could not place caret on ${x}, ${y}`);
+    return;
+  }
+  const { offset, offsetNode } = caretPos;
+  const selection = getWindowSelection();
+  selection.collapse(offsetNode, offset);
+}
+/**
+ * Get the bounding clientRect for the caret.
+ * This also handles the cases where the caret is in a weird place
+ * and therefore has no clientRect by default.
+ * The returned rect should always be interpreted as being in front of the caret.
+ * So when using this to store a location and later resetting it
+ * use the "right" property as the x value.
+ * */
+function getCaretRect(): ClientRect {
```

Comment:
I'm very exited about this bit, this could possibly really enhance our change detection in both the backspace and delete handler. bit worried about some state changes it introduces though, such as introducing  an extra invisibleSpace.

Reply:
yeah I hate that part, but it's the most consistent way I found to get the coordinates when the caret is in a weird position (which happens about 50% of the time)